iml-agent/src/rpm.rs

```diff
@@ -1,25 +1,89 @@
 use regex::Regex;
 
+use std::fmt;
 use std::process::Output;
+use std::result::Result as StdResult;
 
 use crate::{agent_error::ImlAgentError, cmd::cmd_output};
 
-fn parse(output: Output) -> Result<bool, ImlAgentError> {
+enum PackageState {
+    Installed(Version),
+    NotInstalled,
```

Comment:
Did you consider `Option<Version>`?

Reply:
I don't like an option because it doesn't bear domain information. When looking at `Option<Version>` and it's `None`, why is version absent? It's not clear.

Reply:
Well, `PackageState` so far is just renamed `Option`, but I can see below that you parse RPM output, and thus this renaming might make sense if it maps literally to RPM output.

Reply:
I uploaded version with `Option`, but I can't say I like it.
Now in `cli.rs` we have to guess the reason of version absence because it's simply an option. Moreover other errors would probably start occurring as I develop the module and I'll possibly need to get back to `Result` again.

Reply:
Version is absent iff the package is not installed. What other reasons? Of course there could be query error, segfault, and other things that could make us impossible to check, but they are covered by top-level `Result`. No?

Reply:
Do you suggest adding `RpmError` variant to `ImlAgentError` enum?

Reply:
Maybe, there are already `CmdOutputError`, `LiblustreError`.

Reply:
Ok that's an option